rskj-core/src/main/java/co/rsk/core/bc/TransactionPoolImpl.java

```diff
@@ -158,37 +158,49 @@ private RepositorySnapshot getCurrentRepository() {
         return repositoryLocator.snapshotAt(getBestBlock().getHeader());
     }
 
-    @Override
-    public synchronized List<Transaction> addTransactions(final List<Transaction> txs) {
+    private List<Transaction> addSuccesors(Transaction tx) {
```

Comment:
@ajlopez, It seems to me that `addSuccesors` and `getQueuedSuccesor` methods could be merged into single one? So we'd reduce a number of paths through the `queuedTransactions` collection.

Reply:
but `getQueuedSuccesor` is called TWICE from `addSuccesors` in two different logic

Reply:
I guess the current version is pretty sustainable and it's more about refactoring, reducing the amount of code and personal preference )

something like:

```java
        Map<BigInteger, Transaction> txMap = queuedTransactions.getTransactionsWithSender(tx.getSender()).stream()
                .collect(Collectors.toMap(Transaction::getNonceAsInteger, t -> t));

        Transaction nextSuccessor;
        BigInteger nextNonce = tx.getNonceAsInteger().add(BigInteger.ONE);
        while ((nextSuccessor = txMap.get(nextNonce)) != null && internalAddTransaction(nextSuccessor).transactionWasAdded()) {
            added.add(nextSuccessor);
            nextNonce = nextNonce.add(BigInteger.ONE);
        }
```

I also noticed a type in the `getQueuedSuccesor` method name and that `getTransactionsWithSender` cannot return `null`, so no need to check for `null` inside the `getQueuedSuccesor` method.


Inside the `getTransactionsWithSender` method I'd replace `return new ArrayList<>();` onto `Collections.emptyList()`, even though it's not related to the pull request.

Reply:
OK, now using `Collections.emptyList()` and typos fixed, thanks!